### PR TITLE
Azure Virtual Machine Extension module (azure_rm_virtualmachine_extension)

### PR DESCRIPTION
**Pull request by sozercan**
##### SUMMARY
New module that adds support for Azure Virtual Machine Extensions

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine_extension

##### ANSIBLE VERSION
```
ansible 2.4.0 (azure-vmextension 2544f52704) last updated 2017/08/13 12:37:42 (GMT -700)
  config file = None
  configured module search path = [u'/Users/sozercan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/sozercan/Projects/ansible2/lib/ansible
  executable location = ./bin/ansible
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_extension.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                publisher=self.publisher,
+                virtual_machine_extension_type=self.virtual_machine_extension_type,
+                type_handler_version=self.type_handler_version,
+                auto_upgrade_minor_version=self.auto_upgrade_minor_version,
+                settings=self.settings,
+                protected_settings=self.protected_settings
+            )
+            poller = self.compute_client.virtual_machine_extensions.create_or_update(self.resource_group, self.virtual_machine_name, self.name, params)
+            self.get_poller_result(poller)
+        except AzureHttpError as e:
```
Comment (lmazuel) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_extension.py]: should be `CloudError`

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_extension.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            self.log('Error attempting to create the vmextension.')
+            self.fail("Error creating the vmextension: {0}".format(str(e)))
+
+    def delete_vmextension(self):
+        self.log("Deleting vmextension {0}".format(self.name))
+        try:
+            poller = self.compute_client.virtual_machine_extensions.delete(self.resource_group, self.virtual_machine_name, self.name)
+            self.get_poller_result(poller)
+        except AzureHttpError as e:
```
Comment (lmazuel) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_extension.py]: should be `CloudError`

**Review by lmazuel — APPROVED** [review]
Wrong exception used at some places

**Comment by haroldwongms** [comment]
shipit

**Comment by trstringer** [comment]
shipit

**Comment by sozercan** [comment]
bot_status

**Pull request merged**